Rmd/go_pathway_report.Rmd

```diff
@@ -443,23 +444,26 @@ for (i in seq_along(contrasts)) {
 This section uses clusterProfiler `gseKEGG()` to detect enriched KEGG pathways and enrichplot `gseaplot2` to plot running scores.
 
 ```{r 'gsea_analysis_by_contrast_make_plots', collapse=TRUE, eval=params$run_pathway_analysis, include=params$run_pathway_analysis,  warning = FALSE, message = FALSE}
+
 plotListGSEA <- list()
 plotListGSEA_escore <- list()
 contrasts <- levels(allResults$contrast)
+bg_genes <-read.gmt("~/storage/LearnPipline/R-ODAF_Health_Canada/c2.cp.kegg_medicus.v2023.2.Hs.symbols.gmt") 
+
 for (i in seq_along(contrasts)) {
   DEGs_full <- dplyr::left_join(allResults %>% filter(contrast ==  contrasts[i]),
                                 id_table_entrez,
                                 by = "Feature_ID")
   foldChanges <- DEGs_full %>% dplyr::pull(log2FoldChange)
-  names(foldChanges) <- DEGs_full %>% dplyr::pull(Entrez_ID)
+  names(foldChanges) <- DEGs_full %>% dplyr::pull(Gene_Symbol.x)
   foldChanges <- foldChanges %>% sort() %>% rev()
   foldChanges <- foldChanges[!is.na(names(foldChanges))]
   if (length(foldChanges) < 10) {
     plotListGSEA[[i]] <- NA
     plotListGSEA_escore[[i]] <- NA
     next
   }
-  kk <- gseKEGG(foldChanges, keyType = "ncbi-geneid", organism = species_data$kegg_organism)
+  kk <- GSEA(foldChanges, pvalueCutoff = 0.05, TERM2GENE = bg_genes,)
```

Comment:
Yes runs and does not change outputs, allows for the graphs to be included under the go_pathway.html in DEG reports

Reply:
So I have another question about this. You're manually providing the KEGG mapping here. If you look at the documentation for gseKEGG one of the nice things is that species is parameterized and any species on this list can be provided:

https://www.genome.jp/kegg/catalog/org_list.html

This is convenient because we aren't always using human models. So, this change forces users to manually provide a file that they have to dig up somewhere.

Is there no way to get the gseKEGG function to give the right output?